apps/src/lib/ui/SchoolInfoInterstitial.jsx

```diff
@@ -0,0 +1,289 @@
+    borderLeftWidth: 0,
+    borderStyle: 'solid',
+    borderColor: color.lighter_gray,
+  },
+  bottom: {
+    display: 'flex',
+    justifyContent: 'flex-end',
+  },
+  error: {
+    color: color.red,
+  },
+};
+
+const FIREHOSE_EVENTS = {
+  // Interstitial is displayed to the teacher.
+  SHOW: 'show',
+  // Teacher clicked "Save"
+  SUBMIT: 'submit',
+  // School information saved successfully
+  SAVE_SUCCESS: 'save_success',
+  // School information failed to save
+  SAVE_FAILURE: 'save_failure',
+};
+
+export default class SchoolInfoInterstitial extends React.Component {
+  static propTypes = {
+    // This component is tightly bound to the HAML view that renders it and
+    // populates its props, and similarly to the User update API that
+    // it uses to save entered school info.
+    scriptData: PropTypes.shape({
+      formUrl: PropTypes.string.isRequired,
+      authTokenName: PropTypes.string.isRequired,
+      authTokenValue: PropTypes.string.isRequired,
+      existingSchoolInfo: PropTypes.shape({
+        // Note, these names intentionally match the fields on the SchoolInfo
+        // model, not their JavaScript-friendly equivalents.  The mapping
+        // occurs in the constructor, below, and back when we submit.
+        school_id: PropTypes.string,
+        country: PropTypes.string,
+        school_type: PropTypes.string,
+        school_name: PropTypes.string,
+        state: PropTypes.string,
+        zip: PropTypes.string,
+        full_address: PropTypes.string,
+      }).isRequired,
+    }).isRequired,
+    onClose: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    const {existingSchoolInfo} = this.props.scriptData;
+
+    let initialCountry = existingSchoolInfo.country || '';
+    if (initialCountry === 'US') {
+      initialCountry = 'United States';
+    }
+
+    const initialNcesSchoolId = existingSchoolInfo.school_id ?
+      existingSchoolInfo.school_id :
+      (
+        existingSchoolInfo.country === 'United States'
+        &&
+        SCHOOL_TYPES_HAVING_NCES_SEARCH.includes(existingSchoolInfo.school_type)
+        &&
+        (existingSchoolInfo.school_name || existingSchoolInfo.state || existingSchoolInfo.zip)
+      ) ? '-1' : '';
+
+    this.state = {
+      country: initialCountry,
+      schoolType: existingSchoolInfo.school_type || '',
+      schoolName: existingSchoolInfo.school_name || '',
+      schoolState: existingSchoolInfo.state || '',
+      schoolZip: existingSchoolInfo.zip || '',
+      schoolLocation: existingSchoolInfo.full_address || '',
+      ncesSchoolId: initialNcesSchoolId,
+      showSchoolInfoUnknownError: false,
+    };
+  }
+
+  logEvent(eventName, data = {}) {
+    firehoseClient.putRecord('analysis-events', {
+      study: 'school_info_interstitial',
+      study_group: 'control',
+      event: eventName,
+      data_json: JSON.stringify({
+        isComplete: SchoolInfoInterstitial.isSchoolInfoComplete(this.state),
```

Comment:
In addition to `isComplete` would it be useful to record if there is an NCES id too?

Reply:
That makes sense.  Sure!